server/v2/cometbft/abci.go

```diff
@@ -471,9 +471,10 @@ func (c *Consensus[T]) FinalizeBlock(
 	}
 
 	// remove txs from the mempool
-	err = c.mempool.Remove(decodedTxs)
-	if err != nil {
-		return nil, fmt.Errorf("unable to remove txs: %w", err)
+	for _, tx := range decodedTxs {
+		if err = c.mempool.Remove(tx); err != nil {
+			return nil, fmt.Errorf("unable to remove txs: %w", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("unable to remove tx: %w", err)
```
since you are now removing 1:1. I would say we probably dont want to assume removal is synchronous here. how come the change to 1 by 1 instead of batch ?

Reply:
Because of the alignment with types/mempool.Mempool from the SDK.
There is no batch in the interface we have currently.

Reply:
ah dam missed that. then we can keep this

Reply:
we may want to leave a todo to remove concurrently to avoid blocking